src/main/java/com/puppycrawl/tools/checkstyle/filters/SuppressionPatchFilter.java

```diff
@@ -159,16 +175,190 @@ private String getFileName(List<String> singlePatch) {
 
     private List<List<Integer>> getLineRange(Patch<String> patch) {
         final List<List<Integer>> lineRangeList = new ArrayList<>();
+        int totalAddNum = 0;
+        int totalRemoveNum = 0;
         for (int i = 0; i < patch.getDeltas().size(); i++) {
```

Comment:
This for loop should not take more than 15 lines. Please, explain your logic in words. Give as much details as possible. Do this in the issue comment and put here a reference. Let's move discussion into the issue.

Reply:
ok, link is here: https://github.com/checkstyle/patch-filters/issues/63#issuecomment-655266516

Reply:
The logic looks very complex, not maintainable and tricky to test, even checkstyle complains and you ignore those complains which is not good:
```
    <suppress checks="CyclomaticComplexity" files="SuppressionPatchFilter\.java"/>
    <suppress checks="ExecutableStatementCount" files="SuppressionPatchFilter\.java"/>
    <suppress checks="JavaNCSS" files="SuppressionPatchFilter\.java"/>
```
I'm checking your comment in the issue, thanks for details you provided but it might take some time. In the meantime, could you please think:
1. How can you simplify the logic?
2. Are there other ways to update https://github.com/java-diff-utils/java-diff-utils/pull/90/files which would simplify the aforementioned logic?

Reply:
@rdiachenko I simplify the logic using inner class and make a new PR, please review https://github.com/checkstyle/patch-filters/pull/70

Reply:
It is not a simplification. You just hid it within the inner class.

Where are the answers to my two questions above?

Reply:
@rdiachenko I've thought about it, but I haven't come up with a better way to update https://github.com/java-diff-utils/java-diff-utils/pull/90/files.